app/controllers/employment/EmploymentSummaryController.scala

```diff
@@ -90,10 +60,20 @@ class EmploymentSummaryController @Inject()(implicit val mcc: MessagesController
       employmentData match {
         case Seq() if isInYear => Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
         case Seq() if !isInYear => Redirect(AddEmploymentController.show(taxYear))
-        case Seq(employment) if isInYear => status(singleEmploymentSummaryView(taxYear, employment, doExpensesExist))
-        case Seq(employment) if !isInYear => status(singleEmploymentSummaryEOYView(taxYear, employment, yesNoForm))
-        case _ => status(multipleEmploymentsSummaryView(taxYear, employmentData, doExpensesExist, isInYear, yesNoForm))
+        case Seq(employment) if isInYear => Ok(singleEmploymentSummaryView(taxYear, employment, doExpensesExist))
+        case Seq(employment) if !isInYear => Ok(singleEmploymentSummaryEOYView(taxYear, employment, doExpensesExist))
+        case _ => Ok(multipleEmploymentsSummaryView(taxYear, employmentData, doExpensesExist, isInYear))
       }
     }
   }
+
+  def addNewEmployer(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
```

Comment:
Do we need this here? Can't we not redirect directly to EmployerNameController from the view?

Reply:
i dont think so, because the old post method is what creates the new employment id and adds it to session. And then that employment id is whats used through the add an employer flow within the url.

Reply:
That is a bit weird. So we create employment id before we even create the name, meaning if we cancel the operation we have created an Id that is not used. There must be a better way, e.g. create the id when we add the name.
We may need to think about that.

Reply:
Yeah it was done as part of https://github.com/hmrc/income-tax-employment-frontend/pull/82 (https://jira.tools.tax.service.gov.uk/browse/SASS-933) for adding a new employment